packages/sirius-web/frontend/sirius-web-application/src/views/project-browser/list-projects-area/ListProjectsArea.tsx

```diff
@@ -34,34 +33,107 @@ const useListProjectsAreaStyles = makeStyles()((theme) => ({
 }));
 
 export const ListProjectsArea = ({}: ListProjectsAreaProps) => {
+  const { classes } = useListProjectsAreaStyles();
+
   const [state, setState] = useState<ListProjectsAreaState>({
-    page: 0,
-    limit: 20,
+    pageSize: 5,
+    projects: [],
+    hasPrev: false,
+    hasNext: false,
+    startCursor: null,
+    endCursor: null,
+    onRefreshProjects: () => {},
+    previousPageRequired: false,
+    nextPageRequired: false,
   });
 
-  const onPageChange = (page: number) => setState((prevState) => ({ ...prevState, page }));
+  const {
+    data: afterData,
+    loading: afterLoading,
+    refreshProjects: afterRefreshProjects,
+  } = useProjectsAfter(state.endCursor, state.pageSize);
 
-  const { data, refreshProjects } = useProjects(state.page, state.limit);
-  const projects: GQLProject[] = data?.viewer.projects.edges.map((edge) => edge.node) ?? [];
-  const count: number = data?.viewer.projects.pageInfo.count ?? 0;
+  const {
+    data: beforeData,
+    loading: beforeLoading,
+    refreshProjects: beforeRefreshProjects,
+  } = useProjectsBefore(state.startCursor, state.pageSize);
+
+  useEffect(() => {
+    if (state.startCursor === null && state.endCursor === null && !afterLoading && afterData) {
+      setState((prevState) => ({
+        ...prevState,
+        projects: afterData?.viewer.projects.edges.map((edge) => edge.node) ?? [],
+        hasPrev: false,
+        hasNext: afterData?.viewer.projects.pageInfo.hasNextPage ?? false,
+        startCursor: afterData?.viewer.projects.pageInfo.startCursor,
+        endCursor: afterData?.viewer.projects.pageInfo.endCursor,
+        onRefreshProjects: afterRefreshProjects,
+        previousPageRequired: false,
+        nextPageRequired: false,
+      }));
+    } else if (state.previousPageRequired && !beforeLoading && beforeData) {
+      setState((prevState) => ({
+        ...prevState,
+        projects: beforeData?.viewer.projects.edges.map((edge) => edge.node).reverse() ?? [],
```

Comment:
The current implementation of `findAllBefore` returns the projects from the current cursor which is what is asked by the specification.
But it's in the wrong order for the way we want to display them in this component, so I have to reverse the result here.